numpy/__init__.pyi

```diff
@@ -358,6 +358,187 @@ class ndarray(Iterable, Sized, SupportsInt, SupportsFloat, SupportsComplex,
     def __getattr__(self, name) -> Any: ...
 
 
+_Scalar = TypeVar("_Scalar", bound="generic")
+
+class generic:
+    @property
+    def real(self: _Scalar) -> _Scalar: ...
```

Comment:
`real` and `imag` are not guaranteed to return the same scalar type -- they convert complex numbers to reals.